File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.py
class Solution:
    def processStr(self, s: str) -> str:
        result = ""
        hm = {"*": 1, "#": 1, "%": 1}
        for c in s:
            if c in hm:
                if c == '*':
                    result = result[:-1]
                elif c == '#':
                    result += result
                else:
                    result = result[::-1]

            else:
                result += c

        return result                    
            

File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.rs
use std::iter;

impl Solution {
    pub fn process_str(s: String) -> String {
        let mut ret = String::new();
        
        for c in s.chars() {
            if c == '*' { let _ = ret.pop(); }
            else if c == '#' {
                let copy = ret.clone();
                ret.push_str(&copy);
            }
            else if c == '%' {
                ret = ret
                    .chars()
                    .rev()
                    .collect();
            }
            else { 
                println!("{c}");
                ret.push(c);
             }
        }

        ret
    }
}